src/controllers/create-auth-session.controller.spec.js

```diff
@@ -0,0 +1,135 @@
+  let response;
+
+  beforeAll(() => {
+    CreateAuthSession.prototype.handler = jest.fn();
+    jest.useFakeTimers().setSystemTime(systemTime);
+  });
+
+  beforeEach(() => {
+    environment = process.env;
+    controller = new CreateAuthSessionController(new CreateAuthSession());
+
+    request = new TestRequest();
+    response = new TestResponse();
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  afterEach(() => {
+    process.env = environment;
+    jest.clearAllMocks();
+  });
+
+  it('should have an handler method', () => {
+    expect(controller.handler).toBeInstanceOf(Function);
+  });
+
+  it('should throw an error if it receives an invalid service instance', () => {
+    {
+      expect(() => new CreateAuthSessionController()).toThrow('Invalid service instance');
+    }
+    {
+      expect(() => new CreateAuthSessionController(controller)).toThrow('Invalid service instance');
+    }
+  });
+
+  it('should thrown an error if request body is empty', async () => {
+    request.body = {};
+
+    await expect(controller.handler(request, response)).rejects.toBeInstanceOf(z.ZodError);
+  });
+
+  it('should throw an error when any property of the request body is invalid', async () => {
+    {
+      request.body = {
+        email: 'user.one@mail.com',
+      };
+
+      await expect(controller.handler(request, response)).rejects.toBeInstanceOf(z.ZodError);
+    }
+    {
+      request.body = {
+        password: '123456',
+      };
+
+      await expect(controller.handler(request, response)).rejects.toBeInstanceOf(z.ZodError);
+    }
+  });
+
+  it('should calls service with request body data', async () => {
+    const credentials = { email: 'user.one@mail.com', password: '123456' };
+    request.body = credentials;
+
+    await controller.handler(request, response);
+
+    expect(CreateAuthSession.prototype.handler).toBeCalledTimes(1);
+    expect(CreateAuthSession.prototype.handler).toBeCalledWith(credentials);
+  });
+
+  it('should set response token cookie with service return', async () => {
+    {
+      const tokenMock = 'token-mock';
+      const credentials = { email: 'user.one@mail.com', password: '123456' };
+      request.body = credentials;
+
+      CreateAuthSession.prototype.handler.mockResolvedValueOnce(tokenMock);
+
+      await controller.handler(request, response);
+
+      expect(response.cookie).toBeCalledTimes(1);
+      expect(response.cookie).toBeCalledWith('token', tokenMock, {
+        expires: new Date(2022, 9, 30, 12),
+        httpOnly: true,
+        path: '/',
+        secure: false,
+      });
+      expect(response.status).toBeCalledTimes(1);
+      expect(response.status).toBeCalledWith(201);
+      expect(response.json).toBeCalledTimes(1);
+      expect(response.json).toBeCalledWith({ message: 'Successfully authenticated' });
+    }
+    {
+      const tokenMock = 'token-mock';
+      const credentials = { email: 'user.one@mail.com', password: '123456' };
```

Comment:
We can move this to the top scope because this repeats twice with the same values